include/common/types.h

```diff
@@ -424,11 +434,57 @@ struct RefVariant {
   uint64x2_t Data;
 };
 
+struct StrVariant {
+  // Constructors.
+  StrVariant() noexcept { setData(); }
+  StrVariant(const Runtime::Instance::StringInstance *P) noexcept {
+    setData(P);
+  }
+
+  // Getter of type.
+  const ValType &getType() const noexcept {
+    return reinterpret_cast<const ValType &>(toArray()[0]);
+  }
+  ValType &getType() noexcept {
+    return reinterpret_cast<ValType &>(toArray()[0]);
+  }
+
+  // Getter of pointer.
+  Runtime::Instance::StringInstance *getPtr() const noexcept {
+    return reinterpret_cast<Runtime::Instance::StringInstance *>(toArray()[1]);
+  }
+
+  // Check is null.
+  bool isNull() const { return getPtr() == nullptr; }
+
+  // Getter of the raw data.
+  uint64x2_t getRawData() const noexcept { return Data; }
+
+private:
+  // Helper function of converting data to array.
+  const std::array<uint64_t, 2> &toArray() const noexcept {
+    return reinterpret_cast<const std::array<uint64_t, 2> &>(Data);
+  }
+  std::array<uint64_t, 2> &toArray() noexcept {
+    return reinterpret_cast<std::array<uint64_t, 2> &>(Data);
+  }
+
+  // Helper function to set the content.
+  void
+  setData(const Runtime::Instance::StringInstance *Ptr = nullptr) noexcept {
+    getType() = TypeCode::String;
+    toArray()[1] = reinterpret_cast<uintptr_t>(Ptr);
+  }
```

Comment:
Can we use directly `std::string` here?

Reply:
I think the point here is, a component function anyway can only be used

1. by others module. In this case, we have to lowering it, and hence the string immediately get allocated into memory which borrow from a module. We don't have to think about a dual representation.
2. by others component. In this case, that will can only be a plugin, because a user component have to do option 1 (lowering). Therefore, we have to think about a memory passing fully in host environment, which leads we might have to pass data in host raw pointer.

Reply:
This also hints a possible workflow that user component instantiate our plugin with certain values....